packages/berry-shell/tests/shell.test.ts

```diff
@@ -240,4 +240,37 @@ describe(`Simple shell features`, () => {
       stdout: `world\n`,
     });
   });
+
+  it(`should set environment variables`, async () => {
+    await expect(bufferResult(
+      `PORT=1234 node -e 'process.stdout.write(process.env.PORT)'`
+    )).resolves.toMatchObject({
+      stdout: `1234`,
+    });
+  });
+
+  it(`should support setting multiple environment variables`, async () => {
+    await expect(bufferResult(
+      `HOST="localhost" PORT=1234 node -e 'process.stdout.write(process.env.HOST + ":" + process.env.PORT)'`
+    )).resolves.toMatchObject({
+      stdout: `localhost:1234`,
+    });
+  });
+
+  it(`should support setting environment variables with shell interpolation`, async () => {
+    await expect(bufferResult(
+      `HOST="local"$(echo $0) PORT=1234 node -e 'process.stdout.write(process.env.HOST + ":" + process.env.PORT)'`,
+      [`host`])).resolves.toMatchObject({
+      stdout: `localhost:1234`,
+    });
+  });
+
+  ifNotWin32It(`should not parse as env variables when value contains a leading space eg: FOO= bar`, async () => {
```

Comment:
This is unexpected behaviour for me, BASH and consorts clear the variable in this case. I do expect an exception because the command `localhost` can't be found:

```bash
$ HOST= "localhost" PORT=1234 node -e 'process.stdout.write(process.env.HOST + ":" + process.env.PORT)'
bash: localhost: command not found
```